restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -97,7 +123,20 @@ public static RestResponse buildResponse(RoutingResult routingResult, RestLiResp
                                                          .getResourceEntityType();
     if (restLiResponse.hasData() && ResourceEntityType.STRUCTURED_DATA == resourceEntityType)
     {
+      DataSchema dataSchema = null;
+      Class<? extends RecordTemplate> valueClass = routingResult.getResourceMethod().getResourceModel().getValueClass();
+      if (valueClass != null)
+      {
+        dataSchema = DataTemplateUtil.getSchema(valueClass);
+      }
+
       DataMap dataMap = restLiResponse.getDataMap();
+      if (context.getParameters().containsKey(RestConstants.FILL_DEFAULT_VALUE_IN_RESPONSE_PARAM) &&
```

Comment:
Consider moving this logic to the `ResourceContext` interface. This is what I did for `ResourceContext#isReturnEntityRequested` rather than directly looking at the parameters.

Reply:
Yeah that makes the code consistent, thanks